daemon/routing.c

```diff
@@ -551,6 +555,26 @@ static void json_add_route(struct command *cmd,
 	command_success(cmd, null_response(cmd));
 }
 
+void sync_routing_table(struct lightningd_state *dstate, struct peer *peer)
+{
+	struct node *n;
+	struct node_map_iter it;
+	int i;
+	struct node_connection *nc;
+	for (n = node_map_first(dstate->nodes, &it); n; n = node_map_next(dstate->nodes, &it)) {
+		size_t num_edges = tal_count(n->out);
+		for (i = 0; i < num_edges; i++) {
+			nc = n->out[i];
+			if (nc->channel_announcement)
+				queue_pkt_nested(peer, WIRE_CHANNEL_ANNOUNCEMENT, nc->channel_announcement);
+			if (nc->channel_update)
+				queue_pkt_nested(peer, WIRE_CHANNEL_UPDATE, nc->channel_update);
+		}
+		if (n->node_announcement)
+			queue_pkt_nested(peer, WIRE_NODE_ANNOUNCEMENT, n->node_announcement);
+	}
+}
+
```

Comment:
Adding a check for `num_edges` is simple enough, even if we should never have a stored announcement if we have no edges, it just makes this a bit more explicit.